shared/src/main/scala/drt/shared/Api.scala

```diff
@@ -727,13 +747,41 @@ object DataUpdates {
 
 }
 
+class ArrivalsRestorer[A <: WithUnique[UniqueArrival]] {
+  var arrivals: Map[UniqueArrival, A] = Map()
+
+  def removeHashLegacies(theRemoves: Iterable[Int]): Unit = theRemoves.foreach(keyToRemove => arrivals = arrivals.filterKeys(_.legacyUniqueId != keyToRemove))
```

Comment:
I'm not sure what theRemoves means, should it be `theRemovals` or `keysToRemove`